specification/network/resource-manager/Microsoft.Network/stable/2020-04-01/applicationGateway.json

```diff
@@ -593,6 +593,356 @@
+        "responses": {
+          "200": {
+            "description": "Success. The operation returns a list of private link resources on application gateway.",
+            "schema": {
+              "$ref": "#/definitions/ApplicationGatewayPrivateLinkResourceListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/applicationGateways/{applicationGatewayName}/privateEndpointConnections/{connectionName}": {
+      "delete": {
+        "tags": [
+          "ApplicationGatewayPrivateEndpointConnections"
+        ],
+        "operationId": "ApplicationGatewayPrivateEndpointConnections_Delete",
+        "x-ms-examples": {
+          "Delete Application Gateway Private Endpoint Connection": {
+            "$ref": "./examples/ApplicationGatewayPrivateEndpointConnectionDelete.json"
+          }
+        },
+        "description": "Deletes the specified private endpoint connection on application gateway.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "applicationGatewayName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the application gateway."
+          },
+          {
+            "name": "connectionName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the application gateway private endpoint connection."
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Accepted and the operation will complete asynchronously."
+          },
+          "204": {
+            "description": "Request successful. Resource with the specified name does not exist."
+          },
+          "200": {
+            "description": "Delete successful."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
```

Comment:
since we need consistent swagger spec across service. Is there any reason to query the final state from location instead of azure-async-operation? @malopMSFT We already met three kinds of rest calls for delete command. Synchronous delete, asynchronous delete with azure-async-operation header and  asynchronous delete with location header. Should we make it consistent?

Reply:
I think privateLinkService.json is on the same lines for delete privateEndpointConnection.
 
I dont mind changing it to azure-async-operation. but from application gateway perspective, delete of applicationgateway is async-delete with location header. However from usage point of view how are the two different just for my understanding. 

Also can you please share your msft handle. I would like to understand few things.

Reply:
Sure.  I have contacted you in Teams. I just found, for ACR, the swagger spec says they are using azure-async-operation header. However, the real response is using location header.